### ISSUE TITLE
[RFC] Formalize directory structure for including examples files

### ISSUE THREAD

**Issue by ebeahan** [elastic/ecs#1291]
Since introducing the RFC process, it's been found valuable to create additional directories named after the RFC number (`0001`). The folder typically contains proposed schema changes as standalone YAML files, which are useful to pair with the `--include` flag to generate mappings for testing and experimentation. Extended example mappings and source documents, which are too large to fit into the markdown proposal doc reasonably, have also gone into these directories.

The adopted structure can be seen here: https://github.com/elastic/ecs/tree/master/rfcs/text

If we continue this pattern, we should formalize it by adding guidance in the process markdown and template.

**Comment by djptek** [comment]
@ebeahan @kgeller I agree these folders are the perfect place for mappings & source files.

The RFC process as I currently understand requires new field definitions files &/or changes to existing `*.yml` field definitions should also go into these folders, see: https://github.com/elastic/ecs/blob/master/rfcs/PROCESS.md

```
If proposing new fields or changing existing fields, the RFC should also have a corresponding folder (named after the RFC number) in rfcs/text/. The folder should contain the proposed schema changes as standalone YAML files or extended example mappings and larger source documents.
```

When reviewing RFCs in the GitHub GUI there is no immediate way to see the diff between the master and RFC branch, since the file is a copy in a distinct location. The workaround is to run a diff via command line, however, this takes you out of the context of the review process which introduces risk of error and requires adding the PR's remote which increases complexity.

Could we consider proposing changes to field definitions directly in their target field definition files so e.g. adding the change to:

https://github.com/djptek/ecs/blob/rfc_0017_stage_1/schemas/event.yml

rather than copy this to a new location and adding to

https://github.com/djptek/ecs/blob/rfc_0017_stage_1/rfcs/text/0017/event.yml

?

Although not explicitly included in the RFC subfolder, these changes would still be apparent in the GitHub review GUI with the benefit that the GUI simplifies comparison of the delta(s)

Note: if that is what is originally intended, I have misread the RFC process, in which case let me know and I'll instead propose rewording there

**Comment by ebeahan** [comment]
Chatted a bit out-of-band with @kgeller @djptek.

We discussed:

* Mention using`rfcs/text/<folder>` with the `--include` flag, like: `$ python scripts/generator.py --include ./rfcs/text/0012/ --out _testing/rfc-0012`. This use case for `--include` isn't clear right now.
* For new fields, the field definitions should be complete. For changes, the field definitions included in `rfcs/text/<folder>` only need to be a delta ([example](https://github.com/elastic/ecs/blob/master/rfcs/text/0007/user.yml)).
* When removing fields, field defs not required but can propose an updated description of how a field is deprecated and eventually removed.

I see two improvements to build on what was already done in #1336:

1. Also add a mention of using `--include` with `rfcs/text/<RFC>` as another tool for reviewing proposed changes
2. Clarify that field definitions will usually be required for additions or changes but may not be required for removals.

WDYT? Anything else we could capture?

**Comment by djptek** [comment]
@ebeahan thanks for documenting here, that would cover (my) existing doubts. I'll open a PR with the clarifications
